CRI: add docs for AppArmor and Seccomp

feiskyer opened the pull request:
This PR adds docs for AppArmor and Seccomp in CRI.

cc/ @yujuhong @Random-Liu

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34819)

<!-- Reviewable:end -->

feiskyer committed:
CRI: add docs for AppArmor and Seccomp

yujuhong commented:
@timstclair could you help review this? Thanks!

timstclair commented on pkg/kubelet/api/v1alpha1/runtime/api.pb.go:
```diff
@@ -566,7 +566,35 @@ type PodSandboxConfig struct {
+	//      value: see below.
+	//
+	//      key: security.alpha.kubernetes.io/seccomp/container/<container name>
+	//      description: the seccomp profile for the container.
```
Add `(overides pod)`?

feiskyer replied:
ack.

timstclair commented on pkg/kubelet/api/v1alpha1/runtime/api.proto:
```diff
@@ -187,7 +187,35 @@ message PodSandboxConfig {
     map<string, string> labels = 6;
     // Annotations is an unstructured key value map that may be set by external
-    // tools to store and retrieve arbitrary metadata.
+    // tools to store and retrieve arbitrary metadata. Runtimes should support
```
> Runtimes should support...

I think both of these features should be considered optional (nice to haves).

feiskyer replied:
ack

timstclair reviewed (commented):
Docs LGTM, but do we really want to get into the business of enumerating every annotation that the runtime could care about? It sounds like features like these won't be annotations in the near future (discussion [here](https://github.com/kubernetes/kubernetes/issues/30819)), so this might not be a big deal.

feiskyer commented:
> Docs LGTM, but do we really want to get into the business of enumerating every annotation that the runtime could care about? It sounds like features like these won't be annotations in the near future (discussion here), so this might not be a big deal.

It's better to document these annotations in CRI clearly, so runtime maintainers don't need to lookup those annotations from kubernetes API. We can move those features to sandbox/container configs after #30819 .

@timstclair Addressed comments. PTAL.

feiskyer commented:
@k8s-bot cvm gke e2e test this

feiskyer commented:
@timstclair Comments addressed. PTAL.

timstclair commented:
LGTM

feiskyer commented:
@k8s-bot gci gke e2e test this
